docs/source/architecture/data_services_architecture.rst

```diff
@@ -1,4 +1,110 @@
+
+         .. image:: mds_architecture.png
+            :scale: 100 %
+	  		:alt: Motech Data Services Architecture Diagram
+            :align: center
+
+
+        Section following describes motech data services architecture in figure.
+
+MDS UI Layer
+============
+
+#.Schema Editor: It is responsible for creating new EUDEs and also to edit already existing entities (Adding new fields and their lookups).
+#.Data Browser: It is responsible for browsing through entities.
+#.Settings: It is responsible for setting data retention policies .i.e. managing trash and history data. It also provides an option to export and import schema using csv as input or output.
+
+MDS Web
+=======
+
+It acts as controller between MDS UI Layer and MDS Core Layer to view, create and update EUDE and it’s infrastructure classes .i.e services and repositories and to view and update DDE and it’s infrastructure classes .i.e services and repositories. It use pre defined services of MDS.
+
+MDS Core Layer 
+==============
+
+#. MDS Initializer: It is responsible for building all entites in MDS database. It use MDS Constructor for building entites. It’s using TransactionalCallBackWithout Result of Spring framework for configuring Transaction. It includes MDSWeavingHook which is responsible for hooking into OSGi class loading process to dynamically add Jdo Imports and replacing DDE’s with thier bytecodes (extended version) generated by MDS.
+
+Added Jdo Imports are as follows:-
+javax.jdo,javax.jdo.identity,javax.jdo.spi,org.joda.time,org.apache.commons.lang,org.springframework.transaction.support,org.motechproject.mds.filter,org.motechproject.mds.query,org.motechproject.mds.util,org.motechproject.commons.date.util,org.datanucleus.enhancer,org.datanucleus,org.datanucleus.state,org.datanucleus.asm,org.datanucleus.exceptions,org.datanucleus.metadata,org.datanucleus.plugin,org.datanucleus.util,org.datanucleus.identity
+
+
+#. MDS Constructor: It provides methods to create entities, their service implementations and corresponding interfaces.
+
+#. Entity Builder: It is use to build classes of a given entities.
+
+#. Infrastructure Builder: It is responsible for building infrastructure for a given entity .i.e repository, interface and service classes. These classes are created only if not available in the classpath using  javassist to construct them.
+
+#. Metadata Builder: It is responsible for generating JDO metadata for entity.
+
+#. Enum Builder: Value defined in Combo Box/ Field Metadata will be used to create an Enum.
+
+#. Lookup Builder: It allows to create a method without body and added into javassist service interface and method with body added into javassist service class.
+
+#. Processors: Processors process DDEs based on annotations and create Infrastructure and Metadata Information for those entities which can used via DataNucleus enhancer to enhance entities to use them for Persistence Operations.
```

Comment:
Mention that processors use reflections to scan bundles after they are installed (MdsBundleWatcher keeps track of bundles getting installed). The information that is gathered by processors goes to the schema database, which along EUDE information is used by the JarGeneratorService (which delegates much of this to the MDSConstructor) to build the MDS classes, metadata and infrastructure